source/HailPanel.cpp

```diff
@@ -113,13 +113,14 @@ HailPanel::HailPanel(PlayerInfo &player, const shared_ptr<Ship> &ship, function<
 				message += "give you some fuel?";
 			else if(canRepair)
 				message += "patch you up?";
+			SetMessage(message);
 		}
```

Comment:
This looks very strange. Nothing else outside of `SetMessage` modifies the `message` field, but here you are modifying that field directly, then passing a reference to it into `SetMessage` which will then copy that string into `message` again?
Maybe use a local variable to hold the string while it is being prepared and then pass that into `SetMessage`?